pkg/api/meta/scheme_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package meta_test
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api/testapi"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// These types do not follow the list convention as documented in
+// docs/devel/api-convention.md
+var listTypeExceptions = sets.NewString("APIGroupList", "APIResourceList")
+
+func validateListType(target reflect.Type) error {
+	// exceptions
+	if listTypeExceptions.Has(target.Name()) {
+		return nil
+	}
+	if strings.HasSuffix(target.Name(), "List") {
```

Comment:
minor nit, but I suggest you structure this code as

```go
hasListSuffix := strings.HasSuffix(...)
hasListMeta := false
hasItemsField := false
/* loop over fields */
if hasListSuffix != hasItemsField { ... }
```

This way you only need one loop through the fields. Slightly more readable.